store/store_download_test.go

```diff
@@ -1040,3 +1041,36 @@ func (s *storeDownloadSuite) TestTransferSpeedMonitoringWriterUnhappy(c *C) {
 	c.Assert(terr, Equals, true)
 	c.Check(w.Err(), ErrorMatches, "download too slow: .* bytes/sec")
 }
+
+func (s *storeDownloadSuite) TestDownloadTimeoutOnHeaders(c *C) {
+	restore := httputil.MockResponseHeaderTimeout(time.Second)
+	defer restore()
+
+	var mockServer *httptest.Server
+
+	quit := make(chan bool)
+	mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// block the handler, do not send response headers.
+		select {
+		case <-quit:
+		case <-time.After(5 * time.Second):
+			// we expect to hit ResponseHeaderTimeout first
```

Comment:
I'm probably misreading the code, but it seems to me that the ResponseHeaderTimeout happens only after 10 seconds. So, how can this 5 seconds timeout be enough?

Reply:
It's mocked to 1s with httputil.MockResponseHeaderTimeout(time.Second).

Reply:
Ah, I missed that! Nice!